src/msg/async/rdma/RDMAStack.cc

```diff
@@ -462,7 +461,8 @@ void RDMADispatcher::handle_tx_event(ibv_wc *cqe, int n)
 
     // If it's beacon WR, enqueue the QP to be destroyed later
     if (response->wr_id == BEACON_WRID) {
-      enqueue_dead_qp(response->qp_num);
+      std::lock_guard l{lock};
+      enqueue_dead_qp_lockless(response->qp_num);
```

Comment:
It would be better to add a function `enqueue_dead_qp`:

```
void enqueue_dead_qp(uint32_t qpn)
{
   Mutex::Locker l(lock);
   enqueue_dead_qp_lockless(qpn);
}
```

Reply:
good idea